### PR TITLE
Add support for passing `additional_parameters` as supported by Submitit

### LINKED ISSUE #1036: [Feature Request] [Hydra-Submitit-Launcher] Add support for passing `additional_parameters` as supported by Submititi
# 🚀 Feature Request

When calling [Submitit](https://github.com/facebookincubator/submitit) directly (not the plugin), one can pass an optional dictionary argument called [`additional_parameters`](https://github.com/facebookincubator/submitit/blob/e6fc6b2532a2ea480f08747816f14c53f526ef04/submitit/slurm/slurm.py#L432) to add parameters which are not currently available in submitit.

This argument can not be passed when using the submitit-plugin. Example error:

```
Key 'additional_parameters' not in 'SlurmQueueConf'
        full_key: hydra.launcher.additional_parameters
        reference_type=Optional[SlurmQueueConf]
        object_type=SlurmQueueConf
```

We should add support for passing this `additional_parameters` dict via the plugin aswell.

## Motivation

Same as above.

## Pitch

Add support for the dict in [SlurmQueueConf](https://github.com/facebookresearch/hydra/blob/master/plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py#L31) and pass to submitit.AutoExecutor [here](https://github.com/facebookresearch/hydra/blob/dfbebc34020ad3d4bf1ab061565408740e62eb22/plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py#L105)

**Are you willing to open a pull request?** (See [CONTRIBUTING](../../CONTRIBUTING.md)): Yes

## Additional context

Since this change is quite small, is there any reason it wasnt done in the first iteration of the plugin?

### PR DESCRIPTION
**Pull request by shagunsodhani**
Fixes #1036 

## Motivation

Add support for passing `additional_parameters` as supported by Submitit

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

CI
## Related Issues and PRs

#1036

### CODE REVIEW

File: plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py [hunk 1]
```diff
@@ -57,6 +57,9 @@ class SlurmQueueConf(BaseQueueConf):
     # check the following for more info on slurm_max_num_timeout
     # https://github.com/facebookincubator/submitit/blob/master/docs/checkpointing.md
     max_num_timeout: int = 0
+    # Useful to add parameters which are not currently available in the plugin.
```
Comment (omry) [inline plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py:60]: what do you mean by "Useful to add parameters which are not currently available in the plugin"?
what happens when they become available in the plugin?

Comment (shagunsodhani) [reply plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py:60]: If the plugin starts supporting all the slurm parameters, we do not need to have this variable :)

Comment (omry) [reply plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py:60]: ok. I guess we will cross that bridge later, but if for example support for email is added it's not clear what would be the behavior where the email is becoming supported with regard to existing user's configs.

File: news/1036.api_change [hunk 2]
```diff
@@ -0,0 +1 @@
+Support `additional_parameters` as an optional param in the Submitit launcher plugin.
```
Comment (omry) [inline news/1036.api_change:1]: I think `plugin` suffix is more appropriate here.

File: plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py (hunk 1 above)
Comment (shagunsodhani) [reply plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py:60]: Thats a good point.

**Review by omry — COMMENTED** [review]
1. How was this tested?
2. Is this something that can be covered in the tests?

**Comment by shagunsodhani** [comment]
> 1. How was this tested?
> 2. Is this something that can be covered in the tests?

1. Locally :( 
2. Since the change is specific to the slurm-backend, we would need to setup slurm on the CI.

File: news/1036.api_change (hunk 2 above)
Comment (shagunsodhani) [reply news/1036.api_change]: Fixed in https://github.com/facebookresearch/hydra/pull/1054/commits/01b1e4337708bd8d8fd8269cb96fa2aaa55057a5

**Comment by shagunsodhani** [comment]
Will rebase once the tests on master have been fixed

**Comment by shagunsodhani** [comment]
Rebase + force push

**Comment by shagunsodhani** [comment]
Do we plan to backport this to 1.0 ?

**Pull request merged**